src/feature/control/control_hs.c

```diff
@@ -255,6 +262,10 @@ encode_client_auth_cred_for_control_port(
     }
   }
 
+  if (*cred->client_name) {
```

Comment:
Hmm, I think this is naughty but I understand why you did it.

Instead of doing this dereference to understand if a static array is filled up, let's do the following:
Let's change `client_name` to a pointer, and then in `handle_control_onion_client_auth_add()` if a client name is provided we can allocate some memory and fill it in. So now in `encode_client_auth_cred_for_control_port()` we can do the more natural `if (cred->client_name) {`. We will also need to free `client_name` in `client_service_authorization_free_()`. It's a good time to do this now that this struct is not really used so it's easy.